web-services/map-reduce/src/main/java/datawave/webservice/mr/configuration/MapReduceJobConfiguration.java

```diff
@@ -302,6 +316,40 @@ protected void prepareClasspath(String jobId, Job job, Path jobDir) throws Excep
         exportSystemProperties(jobId, job, fs, classpath);
     }
     
+    private File filterJar(File source, List<Pattern> patterns) {
+        File f = null;
+        try {
+            f = File.createTempFile(source.getName() + ".", "");
+            try (FileOutputStream fos = new FileOutputStream(f); ZipOutputStream zipOutputStream = new ZipOutputStream(fos)) {
+                try (ZipInputStream zipInputStream = new ZipInputStream(new FileInputStream(source))) {
+                    for (ZipEntry zipEntry = zipInputStream.getNextEntry(); zipEntry != null; zipEntry = zipInputStream.getNextEntry()) {
+                        boolean include = true;
+                        Iterator<Pattern> patternItr = patterns.iterator();
+                        while (patternItr.hasNext() && include == true) {
+                            Pattern p = patternItr.next();
+                            if (p.matcher(zipEntry.getName()).matches()) {
+                                include = false;
+                            }
+                        }
+                        if (include) {
+                            zipOutputStream.putNextEntry(zipEntry);
+                            byte[] buffer = new byte[2048];
+                            int len;
+                            while ((len = zipInputStream.read(buffer)) > 0) {
+                                zipOutputStream.write(buffer, 0, len);
+                            }
```

Comment:
Another way...
```suggestion
                            ByteStreams.copy(zipInputStream, zipOutputStream);
```

Reply:
Thanks for the suggestions.  This way is much cleaner.  It took me a while to figure out the Zip(Input/Output)Stream semantics and I didn't circle back to clean it up.

Oh the irony that we are using a guava class in the wildfly jvm to remove the wildfly classes from a shaded jar.  I combined both suggestions and pushed a commit.

Reply:
Figured you'd appreciate that. ;)